smart-contracts/contracts/DigitalaxAuction.sol

```diff
@@ -223,27 +267,40 @@ contract DigitalaxAuction is Context, ReentrancyGuard {
         address winner = highestBid.bidder;
         uint256 winningBid = highestBid.bid;
 
-        // Ensure there is a winner
-        require(winner != address(0), "DigitalaxAuction.resultAuction: No one has bid");
+        // Ensure auction not already resulted
+        require(winningBid >= auction.reservePrice, "DigitalaxAuction.resultAuction: reserve not reached");
 
-        // FIXME check to ensure reserve is met
+        // Ensure there is a winner
+        require(winner != address(0), "DigitalaxAuction.resultAuction: no open bids");
 
-        // Clean up the highest winner
+        // Clean up the highest bid
         delete highestBids[_garmentTokenId];
 
-        // FIXME delete auction too (maybe)?
-
-        // TODO check this feature against the spec? Does this need to be on-chain?
         // Record the primary sale price for the garment
         garmentNft.setPrimarySalePrice(_garmentTokenId, winningBid);
 
-        // Send the designer their earnings
-        (bool designerTransferSuccess,) = garmentNft.garmentDesigners(_garmentTokenId).call{value : winningBid}("");
-        require(designerTransferSuccess, "DigitalaxAuction.resultAuction: Failed to send the designer their royalties");
+        if (winningBid > auction.reservePrice) {
+            // Work out total above the reserve
+            uint256 aboveReservePrice = winningBid.sub(auction.reservePrice);
+
+            // Work out platform fee from above reserve amount
+            uint256 platformFeeAboveReserve = (aboveReservePrice.div(1000)).mul(platformFee);
+
+            // Send platform fee
+            (bool platformTransferSuccess,) = platformFeeRecipient.call{value : platformFeeAboveReserve}("");
+            require(platformTransferSuccess, "DigitalaxAuction.resultAuction: Failed to send platform fee");
+
+            // Send remaining to designer
+            (bool designerTransferSuccess,) = garmentNft.garmentDesigners(_garmentTokenId).call{value : winningBid.sub(platformFeeAboveReserve)}("");
+            require(designerTransferSuccess, "DigitalaxAuction.resultAuction: Failed to send the designer their royalties");
+        } else {
+            // Send all to the designer
+            (bool designerTransferSuccess,) = garmentNft.garmentDesigners(_garmentTokenId).call{value : winningBid}("");
+            require(designerTransferSuccess, "DigitalaxAuction.resultAuction: Failed to send the designer their royalties");
+        }
```

Comment:
would never enter here?

Reply:
nope, is reserve meet directly it goes down this path

Reply:
So this just saves gas I guess